### PR TITLE
feat(performance): concurrent engine scans by platform

### LINKED ISSUE #3085: Enhance KICS Performance
### Description
Enhance KICS performance by running scans concurrently by parser

### PR DESCRIPTION
**Pull request by joaoReigota1**
Signed-off-by: João Reigota <joao.reigota@checkmarx.com>

Closes  #3085

**Proposed Changes**
- Run kics concurrently by platform
- changed progress bar location
- changed parser build to return an array of parsers 
- Changed Metrics Locations to prevent errors

I submit this contribution under the Apache-2.0 license.

### CODE REVIEW

File: pkg/scanner/scanner.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	}()
+
+	select {
+	case <-wgDone:
+		break
+	case err := <-errCh:
+		close(errCh)
+		return err
+	}
+	metrics.Metric.Stop()
```
Comment (felipe-avelar) [inline pkg/scanner/scanner.go:50]: What happens if an error occurs? I think this could be executed on `defer`

**Review by felipe-avelar — CHANGES_REQUESTED** [review]
Beside the single comment, it looks good to me

**Review by felipe-avelar — DISMISSED** [review]
LGTM

**Review by rogeriopeixotocx — APPROVED** [review]
LCTM

**Pull request merged**